pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -223,143 +261,167 @@ func TestAmbientIndex(t *testing.T) {
 	assert.Equal(t,
-		controller.ambientIndex.Lookup("testnetwork/127.0.0.3")[0].Address.GetWorkload().Waypoint.GetAddress().Address,
+		s.lookup(s.addr("127.0.0.3"))[0].Address.GetWorkload().Waypoint.GetAddress().Address,
 		netip.MustParseAddr("10.0.0.2").AsSlice())
+
 	// As should workload via Service
 	assert.Equal(t,
-		controller.ambientIndex.Lookup("testnetwork/10.0.0.1")[1].Address.GetWorkload().Waypoint.GetAddress().Address,
+		s.lookup(s.addr("10.0.0.1"))[1].Address.GetWorkload().Waypoint.GetAddress().Address,
 		netip.MustParseAddr("10.0.0.2").AsSlice())
 
-	addPods("127.0.0.201", "waypoint2-sa", "waypoint-sa",
+	s.addPods(t, "127.0.0.201", "waypoint2-sa", "waypoint-sa",
 		map[string]string{constants.ManagedGatewayLabel: constants.ManagedGatewayMeshControllerLabel},
-		map[string]string{constants.WaypointServiceAccount: "sa2"})
-	assertEvent(t, fx, "cluster0//Pod/ns1/waypoint2-sa")
+		map[string]string{constants.WaypointServiceAccount: "sa2"}, true, corev1.PodRunning)
+	s.assertEvent(t, s.pod("waypoint2-sa"))
 	// Unrelated SA should not change anything
 	assert.Equal(t,
-		controller.ambientIndex.Lookup("testnetwork/127.0.0.3")[0].Address.GetWorkload().Waypoint.GetAddress().Address,
+		s.lookup(s.addr("127.0.0.3"))[0].Address.GetWorkload().Waypoint.GetAddress().Address,
 		netip.MustParseAddr("10.0.0.2").AsSlice())
 
 	// Adding a new pod should also see the waypoint
-	addPods("127.0.0.6", "name6", "sa1", map[string]string{"app": "a"}, nil)
-	assertEvent(t, fx, "cluster0//Pod/ns1/name6")
+	s.addPods(t, "127.0.0.6", "pod6", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
+	s.assertEvent(t, s.pod("pod6"))
 	assert.Equal(t,
-		controller.ambientIndex.Lookup("testnetwork/127.0.0.6")[0].Address.GetWorkload().Waypoint.GetAddress().Address,
+		s.lookup(s.addr("127.0.0.6"))[0].Address.GetWorkload().Waypoint.GetAddress().Address,
 		netip.MustParseAddr("10.0.0.2").AsSlice())
 
-	deletePod(t, pc, "name6")
-	assertEvent(t, fx, "cluster0//Pod/ns1/name6")
+	s.deletePod(t, "pod6")
+	s.assertEvent(t, s.pod("pod6"))
 
-	deletePod(t, pc, "name3")
-	assertEvent(t, fx, "cluster0//Pod/ns1/name3")
-	deletePod(t, pc, "name2")
-	assertEvent(t, fx, "cluster0//Pod/ns1/name2")
+	s.deletePod(t, "pod3")
+	s.assertEvent(t, s.pod("pod3"))
+	s.deletePod(t, "pod2")
+	s.assertEvent(t, s.pod("pod2"))
 
-	deleteService(t, sc, "waypoint-ns")
-	assertEvent(t, fx, "cluster0//Pod/ns1/name1",
-		"cluster0//Pod/ns1/waypoint-ns-pod",
-		"ns1/waypoint-ns.ns1.svc.company.com",
+	s.deleteService(t, "waypoint-ns")
+	s.assertEvent(t, s.pod("pod1"),
+		s.pod("waypoint-ns-pod"),
+		s.svc("waypoint-ns"),
 	)
 
 	assert.Equal(t,
-		controller.ambientIndex.Lookup("testnetwork/10.0.0.1")[1].Address.GetWorkload().Waypoint,
+		s.lookup(s.addr("10.0.0.1"))[1].Address.GetWorkload().Waypoint,
 		nil)
+}
+
+// TODO(nmittler): Consider splitting this into multiple, smaller tests.
+func TestAmbientIndex_Policy(t *testing.T) {
```

Comment:
This was the remainder of the original `TestAmbientIndex`. It's all policy related. I tried finding a way to break it up, but it's a bit entagled and I lost interest :).